nixos/modules/config/shells-environment.nix

```diff
@@ -42,8 +42,8 @@ in
         strings.  The latter is concatenated, interspersed with colon
         characters.
       '';
-      type = with types; attrsOf (oneOf [ (listOf str) str path ]);
-      apply = mapAttrs (n: v: if isList v then concatStringsSep ":" v else "${v}");
+      type = with types; attrsOf (oneOf [ (listOf (oneOf [ float int str ])) float int str path ]);
+      apply = mapAttrs (n: v: if isList v then concatStringsSep ":" v else toString v);
```

Comment:
`concatStringSep` cannot handle a list of float or int.

Reply:
I must admit, I didn't test that explicitly, only plain values :/

Reply:
I've added a map to fix that.

Reply:
`concatMapStringsSep`?

Reply:
I don't get your question

Reply:
> I don't get your question

`lib.concatMapStringsSep ":" toString [ 1 2 3 true false "test" ]`